IntegrationTests/test_runner.py

```diff
@@ -0,0 +1,370 @@
+if "CLASSPATH" not in os.environ:  # REQUIRED
+    sys.exit("Please set the CLASSPATH environment variable")
+
+if "EXECUTABLE" not in os.environ:  # REQUIRED
+    sys.exit("Please set the EXECUTABLE environment variable")
+
+if "TEST_EXCEPTIONS" in os.environ:  # OPTIONAL
+    external_vars.TEST_EXCEPTIONS = os.environ["TEST_EXCEPTIONS"]
+
+if "GENERATE_REPORT" in os.environ:  # OPTIONAL
+    # Value is the location
+    # Its prescense in env variables signifies intent to save
+    external_vars.GENERATE_REPORT = os.environ["GENERATE_REPORT"]
+
+external_vars.CLASSPATH = os.environ["CLASSPATH"]
+external_vars.EXECUTABLE = os.environ["EXECUTABLE"]
+
+if external_vars.TEST_EXCEPTIONS:
+    with open(f"{external_vars.TEST_EXCEPTIONS}", "r", encoding="utf-8") as file:
+        yamlFile = yaml.safe_load(file)
+
+        if "known_failures" in yamlFile.keys():
+            external_vars.known_failures = yamlFile["known_failures"]
+            if external_vars.known_failures is None:
+                external_vars.known_failures = []
+        else:
+            external_vars.known_failures = []
+
+        if "failing_as_unspecified" in yamlFile.keys():
+            external_vars.failing_as_unspecified = yamlFile["failing_as_unspecified"]
+            if external_vars.failing_as_unspecified is None:
+                external_vars.failing_as_unspecified = []
+        else:
+            external_vars.failing_as_unspecified = []
+
+        if "unsupported" in yamlFile.keys():
+            external_vars.unsupported = yamlFile["unsupported"]
+            if external_vars.unsupported is None:
+                external_vars.unsupported = []
+        else:
+            external_vars.unsupported = []
+
+        if "do_not_run" in yamlFile.keys():
+            external_vars.do_not_run = yamlFile["do_not_run"]
+            if external_vars.do_not_run is None:
+                external_vars.do_not_run = []
+        else:
+            external_vars.do_not_run = []
+
+
+def prepare_tests():
+    """
+    Prepare all of the tests and their relevent information into a dictionary
+    so that the test runner understands each test
+    """
+    test_files = []
+    read_directory(location, test_files)
+    test_files = sorted(test_files)
+    return collect_tests(test_files)
+
+
+@pytest.mark.parametrize(
+    "name,stdout,stderr,custom_classpath,case_sensitive",
+    prepare_tests(),
+    ids=[str(test_args[0]) for test_args in prepare_tests()],
+)
+# pylint: disable=too-many-branches
+def tests_runner(name, stdout, stderr, custom_classpath, case_sensitive):
+    """
+    Take an array of dictionaries with test file names and expected output
+    Run all of the tests and check the output
+    Cleanup the build directory if required
+    """
+
+    # Check if a test shoudld not be ran
+    if str(name) in external_vars.do_not_run:
+        pytest.skip("Test included in do_not_run")
+
+    if custom_classpath != "None":
```

Comment:
Hm, why is this a string "None", instead of just a `None`? 
I think there's a bug else where.